renku/cli/workflow.py

```diff
@@ -111,6 +111,55 @@
 values YAML file and passing that using ``--values``. Provider specific
 settings can be passed as file using the ``--config`` parameter.
 
+Iterate Plans
+*************
+
+.. image:: ../_static/asciicasts/iterate_plan.gif
+   :width: 850
+   :alt: Iterate Plans
+
+For executing a Plan with different parameterization ``renku workflow iterate``
+could be used. This sub-command is basically conducting a 'grid search'-like
+execution of a Plan, with parameter-sets provided by the user.
+
+.. code-block:: console
+
+    $ renku workflow iterate --map parameter-1=[1,2,3] \
+            --map parameter-2=[10,20] my-run
+
+The set of possible values for a parameter can be given by ``--map`` command
+line argument or or by specifying them in a values YAML file and passing that
+using ``--mapping``.
+
+By default ``renku workflow iterate`` will execute all the combination of the
+given parameters' list of possible values. Sometimes it is desired that instead
+of all the combination of possible values, a specific tuple of values are
+executed. This could be done by marking the parameters that should be bind
+together with the ``@tag`` suffix in their names.
+
+.. code-block:: console
+
+    $ renku workflow iterate --map parameter-1@tag1=[1,2,3] \
+            --map parameter-1@tag1=[10,5,30] my-run
+
+This will result in only three distinct execution of the ``my-run`` Plan,
+with the following parameter combinations: ``[(1,10), (2,5), (3,30)]``. It is
+important to note, that parameters that have the same tag, should have the same
+number of possible values, i.e. the values list should have the same length.
+
+There's a special template variable for parameter values ``{iter_index}``, which
+can be used to mark each iteration's index in a value of a parameter. The template
+variable is going to be substituted with the iteration index (0, 1, 2, ...).
+
+.. code-block:: console
+
+    $ renku workflow iterate --map parameter-1@tag1=[10,20,30] \
```

Comment:
```suggestion
    $ renku workflow iterate --map parameter-1=[10,20,30] \
```
tag is not relevant in this example, right?